luarules/gadgets/unit_collision_damage_behavior.lua

```diff
@@ -0,0 +1,130 @@
+function gadget:GetInfo()
+	return {
+		name = "Collision Damage Behavior",
+		desc = "Magnifies the default engine ground and object collision damage and handles max impulse limits",
+		author = "SethDGamre",
+		date = "2024.8.29",
+		license = "GNU GPL, v2 or later",
+		layer = 0,
+		enabled = true
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+--use customparams.fall_damage_multiplier = <number> a multiplier that's applied to defaultDamageMultiplier.
+local fallDamageMagnificationFactor = 14 --the multiplier by which default engine ground collision damage is multiplied. change this value to reduce the amount of fall/collision damage taken for all units.
+
+local groundCollisionDefID = Game.envDamageTypes.GroundCollision
+local objectCollisionDefID = Game.envDamageTypes.ObjectCollision
+local objectCollisionVelocityThreshold = 3.3 --this defines how fast a unit has to be moving in order to take object collision damage
+local maxImpulseThreshold = 0.08 --incease this to make units move less from impulse. This defines the max impulse damage a unit can take expressed in a percentage multiplier
+local cooldownsFrameThreshold = Game.gameSpeed --this is how long in frames before a unit's maximum impulse limit resets.
```

Comment:
Yes. This defines how long a unit has to wait before its Max impule received is reset.
This prevents infrequent space launches in a performant way. 
It also gives the game balance team a method to prevent stun locking of low acceleration units by basically giving it i-frames against impulse